erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -405,160 +361,24 @@ void WebRtcConnection::onCandidate(const CandidateInfo& cand, Transport *transpo
-  RtcpHeader *chead = reinterpret_cast<RtcpHeader*> (buf);
-  if (!chead->isRtcp()) {
-    uint32_t recvSSRC = head->getSSRC();
-    if (isVideoSourceSSRC(recvSSRC)) {
-      packet->type = VIDEO_PACKET;
-    } else if (isAudioSourceSSRC(recvSSRC)) {
-      packet->type = AUDIO_PACKET;
-    }
-  }
-
-  if (!pipeline_initialized_) {
-    ELOG_DEBUG("%s message: Pipeline not initialized yet.", toLog());
+  if (getCurrentState() != CONN_READY) {
     return;
   }
-
-  pipeline_->read(std::move(packet));
+  media_stream_->onTransportData(packet, transport);
 }
 
-void WebRtcConnection::read(std::shared_ptr<DataPacket> packet) {
-  char* buf = packet->data;
-  int len = packet->length;
-  // PROCESS RTCP
-  RtpHeader *head = reinterpret_cast<RtpHeader*> (buf);
-  RtcpHeader *chead = reinterpret_cast<RtcpHeader*> (buf);
-  uint32_t recvSSRC;
-  if (!chead->isRtcp()) {
-    recvSSRC = head->getSSRC();
-  } else if (chead->packettype == RTCP_Sender_PT) {  // Sender Report
-    recvSSRC = chead->getSSRC();
-  }
-
-  // DELIVER FEEDBACK (RR, FEEDBACK PACKETS)
-  if (chead->isFeedback()) {
-    if (fb_sink_ != nullptr && shouldSendFeedback_) {
-      fb_sink_->deliverFeedback(std::move(packet));
-    }
-  } else {
-    // RTP or RTCP Sender Report
-    if (bundle_) {
-      // Check incoming SSRC
-      // Deliver data
-      if (isVideoSourceSSRC(recvSSRC)) {
-        parseIncomingPayloadType(buf, len, VIDEO_PACKET);
-        video_sink_->deliverVideoData(std::move(packet));
-      } else if (isAudioSourceSSRC(recvSSRC)) {
-        parseIncomingPayloadType(buf, len, AUDIO_PACKET);
-        audio_sink_->deliverAudioData(std::move(packet));
-      } else {
-        ELOG_DEBUG("%s unknownSSRC: %u, localVideoSSRC: %u, localAudioSSRC: %u",
-                    toLog(), recvSSRC, this->getVideoSourceSSRC(), this->getAudioSourceSSRC());
-      }
-    } else {
-      if (packet->type == AUDIO_PACKET && audio_sink_ != nullptr) {
-        parseIncomingPayloadType(buf, len, AUDIO_PACKET);
-        // Firefox does not send SSRC in SDP
-        if (getAudioSourceSSRC() == 0) {
-          ELOG_DEBUG("%s discoveredAudioSourceSSRC:%u", toLog(), recvSSRC);
-          this->setAudioSourceSSRC(recvSSRC);
-        }
-        audio_sink_->deliverAudioData(std::move(packet));
-      } else if (packet->type == VIDEO_PACKET && video_sink_ != nullptr) {
-        parseIncomingPayloadType(buf, len, VIDEO_PACKET);
-        // Firefox does not send SSRC in SDP
-        if (getVideoSourceSSRC() == 0) {
-          ELOG_DEBUG("%s discoveredVideoSourceSSRC:%u", toLog(), recvSSRC);
-          this->setVideoSourceSSRC(recvSSRC);
-        }
-        // change ssrc for RTP packets, don't touch here if RTCP
-        video_sink_->deliverVideoData(std::move(packet));
-      }
-    }  // if not bundle
-  }  // if not Feedback
-}
-
-void WebRtcConnection::notifyToEventSink(MediaEventPtr event) {
-  event_sink_->deliverEvent(event);
+int WebRtcConnection::sendPLI() {
```

Comment:
who is using this sendPLI now? shouldn't we call mediaStream's sendPLI directly?